### PR TITLE
chore: testing artifact caching proxy

### PR DESCRIPTION
**Pull request by lemeurherve**
Testing PR https://github.com/jenkins-infra/pipeline-library/pull/502

Ref: https://github.com/jenkins-infra/helpdesk/issues/2752

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@Library('pipeline-library@refs/pull/458')
```
Comment (lemeurherve) [inline Jenkinsfile:1]:
```suggestion
@Library('pipeline-library@pulls/458/head')

```

File: Jenkinsfile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@Library('pipeline-library@pulls/458/head')
```
Comment (smerle33) [inline Jenkinsfile:1]: shouldn't it be with a _
@Library('pipeline-library@pulls/458/head') _

Comment (lemeurherve) [reply Jenkinsfile:1]:
```suggestion
@Library('pipeline-library@pulls/458/head') _
```

File: Jenkinsfile [hunk 3]
```diff
@@ -1,5 +1,7 @@
+@Library('pipeline-library@pull/499/head') _
+
 /*
  See the documentation for more options:
  https://github.com/jenkins-infra/pipeline-library/
 */
-buildPlugin(useContainerAgent: true, platforms: ['linux'], jdkVersions: ['8','11','17'])
+buildPlugin(useContainerAgent: false, platforms: ['linux', 'windows'], artifactCachingProxyEnabled: true, jdkVersions: ['8','11','17'])
```
Comment (jglick) [inline Jenkinsfile:7]: `useContainerAgent` also tested?

Comment (lemeurherve) [reply Jenkinsfile:7]: @jglick successfully tested [with a replay (Linux only for now)](https://ci.jenkins.io/job/Plugins/job/jenkins-infra-test-plugin/job/PR-44/128):
```
@Library('pipeline-library@pull/499/head') _

/*
 See the documentation for more options:
 https://github.com/jenkins-infra/pipeline-library/
*/
buildPlugin(useContainerAgent: true, platforms: ['linux'], artifactCachingProxyEnabled: true, jdkVersions: ['11'])

```

Comment (lemeurherve) [reply Jenkinsfile:7]: Testing my new PR on both, looks good.

**Pull request merged**